PWGHF/HFC/TableProducer/correlatorLcHadrons.cxx

```diff
@@ -0,0 +1,700 @@
+using SelectedTracksMcGen = soa::Filtered<aod::McParticles>;
+
+// Code to select collisions with at least one Lambda_c
+struct HfCorrelatorLcHadronsSelection {
+  Produces<aod::LcSelection> lcSel;
+
+  Configurable<int> selectionFlagLc{"selectionFlagLc", 1, "Selection Flag for Lc"};
+  Configurable<float> yCandMax{"yCandMax", 0.8, "max. cand. rapidity"};
+  Configurable<float> ptCandMin{"ptCandMin", 1., "min. cand. pT"};
+
+  HfHelper hfHelper;
+  SliceCache cache;
+
+  Partition<soa::Join<aod::HfCand3Prong, aod::HfSelLc>> selectedLcCandidates = aod::hf_sel_candidate_lc::isSelLcToPKPi >= selectionFlagLc || aod::hf_sel_candidate_lc::isSelLcToPiKP >= selectionFlagLc;
+  Partition<soa::Join<aod::HfCand3Prong, aod::HfSelLc, aod::HfCand3ProngMcRec>> selectedLcCandidatesMc = aod::hf_sel_candidate_lc::isSelLcToPKPi >= selectionFlagLc || aod::hf_sel_candidate_lc::isSelLcToPiKP >= selectionFlagLc;
+
+  void processLcSelectionData(aod::Collision const& collision,
+                              soa::Join<aod::HfCand3Prong, aod::HfSelLc> const& candidates)
+  {
+    bool isLcFound = 0;
+    if (selectedLcCandidates.size() > 0) {
+      auto selectedLcCandidatesGrouped = selectedLcCandidates->sliceByCached(aod::hf_cand::collisionId, collision.globalIndex(), cache);
+
+      for (const auto& candidate : selectedLcCandidatesGrouped) {
+        if (!TESTBIT(candidate.hfflag(), aod::hf_cand_3prong::DecayType::LcToPKPi)) {
+          continue;
+        }
+        if (yCandMax >= 0. && std::abs(hfHelper.yLc(candidate)) > yCandMax) {
+          continue;
+        }
+        if (ptCandMin >= 0. && candidate.pt() < ptCandMin) {
+          continue;
+        }
+        isLcFound = 1;
+        break;
+      }
+    }
+    lcSel(isLcFound);
+  }
+  PROCESS_SWITCH(HfCorrelatorLcHadronsSelection, processLcSelectionData, "Process Lc Collision Selection Data", true);
+
+  void processLcSelectionMcRec(aod::Collision const& collision,
+                               soa::Join<aod::HfCand3Prong, aod::HfSelLc, aod::HfCand3ProngMcRec> const& candidates)
+  {
+    bool isLcFound = 0;
+    if (selectedLcCandidatesMc.size() > 0) {
+      auto selectedLcCandidatesGroupedMc = selectedLcCandidatesMc->sliceByCached(aod::hf_cand::collisionId, collision.globalIndex(), cache);
+      for (const auto& candidate : selectedLcCandidatesGroupedMc) {
+        // check decay channel flag for candidate
+        if (!TESTBIT(candidate.hfflag(), aod::hf_cand_3prong::DecayType::LcToPKPi)) {
+          continue;
+        }
+        if (yCandMax >= 0. && std::abs(hfHelper.yLc(candidate)) > yCandMax) {
+          continue;
+        }
+        if (ptCandMin >= 0. && candidate.pt() < ptCandMin) {
+          continue;
+        }
+        isLcFound = 1;
+        break;
+      }
+    }
+    lcSel(isLcFound);
+  }
+  PROCESS_SWITCH(HfCorrelatorLcHadronsSelection, processLcSelectionMcRec, "Process Lc Selection McRec", false);
+
+  void processLcSelectionMcGen(aod::McCollision const& mcCollision,
```

Comment:
What's the reason to keep MC gen and MC reco separated in two process functions? Do you need only one of the two in some cases?